### ISSUE TITLE
Windows: support resource compilation

### ISSUE THREAD

**Issue by laszlocsomor** [bazelbuild/bazel#8273]
### Description of the problem / feature request:

Support compiling .rc files on Windows. The resource compiler is part of the Windows SDK.

### Feature requests: what underlying problem are you trying to solve with this feature?

To support compiling resources (icons, string tables, version info, etc.) on Windows to object files, and linking them into C++ binaries.

### What operating system are you running Bazel on?

Windows

### What's the output of `bazel info release`?

0.25.0

**Comment by laszlocsomor** [comment]
Important to know: the resource compiler (rc.exe) is in the Windows SDK, not in MSVC.

**Comment by laszlocsomor** [comment]
Useful info from @meteorcloudy : https://github.com/bazelbuild/bazel/commit/a3dd7775d2dbf7e4368712626e11c15be7a23efd#diff-457360df3a98d4f91f49ccc5c8f3b35a has exposed functions we could use to find the Windows SDK and the resource compiler within.

**Comment by laszlocsomor** [comment]
We could put the Starlark rule into https://github.com/bazelbuild/rules_cc.

**Comment by laszlocsomor** [comment]
@meteorcloudy : could you briefly explain how to fix this? I think you have the most expertise.
I understand windows_cc_configure now knows where the Windows SDK is (https://github.com/bazelbuild/bazel/commit/8167f46ce53201ca3d725b6262c6b9e31dac2f13), but how to go from there? Is the SDK path recorded in the cc toolchain somehow? How would a custom rule find out the SDK path (and thus the rc.exe path)?

**Comment by meteorcloudy** [comment]
This is actually an internal rule we could open source to rules_cc, search for `msvc/build_windows.bzl`

**Comment by laszlocsomor** [comment]
Status:
- rule implementation PR https://github.com/bazelbuild/bazel/pull/8907 still pending
- Bazel-side pre-work already upstreamed (will be in Bazel 1.0)
- rule will eventually be in https://github.com/bazelbuild/rules_cc, outside of Bazel core, so this bug no longer will be Bazel-specific

**Comment by laszlocsomor** [comment]
Removing 1.0 label because this rule will be out of Bazel core.

**Comment by ivucica** [comment]
Is this meant to be in 2.0.0?

```
$ bazel query @bazel_tools//src/main/res:win_res.bzl
Loading: 0 packages loaded
ERROR: no such package '@bazel_tools//src/main/res': BUILD file not found in directory 'src/main/res' of external repository @bazel_tools. Add a BUILD file to a directory to mark it as a package.
Loading: 0 packages loaded
Loading: 0 packages loaded
```

and

```
$ bazel build //:app_resources
Loading:
Loading: 0 packages loaded
ERROR: Skipping '//:app_resources': error loading package '': Label '@bazel_tools//src/main/res:win_res.bzl' is invalid because 'src/main/res' is not a package; perhaps you meant to put the colon here: '@bazel_tools//src:main/res/win_res.bzl'?
WARNING: Target pattern parsing failed.
ERROR: error loading package '': Label '@bazel_tools//src/main/res:win_res.bzl' is invalid because 'src/main/res' is not a package; perhaps you meant to put the colon here: '@bazel_tools//src:main/res/win_res.bzl'?
INFO: Elapsed time: 0.131s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (1 packages loaded)
FAILED: Build did NOT complete successfully (1 packages loaded)
```

Looks like `~/_bazel_${USER}/install/*/embedded_tools/src/main` contains just `cpp`, `native` and `protobuf`. I understand this is meant to go into `rules_cc`, but until it goes there, can the rule be released anyway? As it is, it's not very useful.

**Comment by laszlocsomor** [comment]
No, it's not meant to be in `@bazel_tools` in 2.0.0. The rule only exists in Bazel's source tree.

I understand that the rule isn't generally useful in its current state, but due to changed priorities I'm no longer planning to move it to `rules_cc`, sorry.

@ivucica , I recommend copying the rule to your source tree. It should work. If you (or someone else) wants to move it to `rules_cc` for everyone's benefit, then please send a PR to `rules_cc`.

**Comment by jheaff1** [comment]
Hi,

Is there any update on this? It appears to me that neither `@rules_cc` nor `@bazel_tools` provide support for resource compilation.

**Comment by laszlocsomor** [comment]
Sorry, no updates from me.

@meisterT , who's working on Windows support these days?

**Comment by meisterT** [comment]
cc @meteorcloudy @mai93

**Comment by meteorcloudy** [comment]
I think rules_cc is the right place to go.  @oquenchil do you see any concerns?

/cc @andrewkatson might could help with this.

**Comment by oquenchil** [comment]
No one is really working on Windows support these days unfortunately.

**Comment by cpsauer** [comment]
While we're all here, I'd love to get a sense for the overall state of Windows dev on Bazel.

Both from on the user experience side and in terms of Bazel team priorities.
 
[@jheaff1 are things working well, and this is the feature you'd most want? Or one of several issues stopping you from adopting?]

**Comment by jheaff1** [comment]
It’s a shame to hear that there is no current development for Bazel on Windows. 

My need for Windows Resource Compilation support is that I would like to use `rules_foreign_cc` to build OpenSSL on Windows. The Windows build of OpenSSL requires that the resource compiler (`rc.exe`) is on the PATH or the RC environment variable is set to the location of the `rc.exe`. With the functionally described in this PR moved to `@bazel_tools` or `rules_cc`, the location of `rc.exe` can easily be accessed in rule implementations.

**Comment by meteorcloudy** [comment]
I want to clarify a bit. Yes, Windows development is current relatively lower priority for the Bazel team, because there is no obviously feature parity missing and new features we develop are also tested for Windows. But we do still look into user reported Windows issues and try our best to solve them.

FYI, I recently wrote a doc about our overall Bazel Windows support, check [here](https://docs.google.com/document/d/17YIqUdffxpwcKP-0whHM6TFELN8VohTpjiiEIbbRfts/edit?usp=sharing) if you are interested.

> The Windows build of OpenSSL requires that the resource compiler (rc.exe) is on the PATH or the RC environment variable is set to the location of the rc.exe.

As for this specific issue, I think you can workaround by copying the rule that locates the `rc.exe` compiler. We'll look into how to merge this rc compiling rule into rules_cc as soon as possible.

**Comment by dizcza** [comment]
> I have submitted a PR to `rules_cc` to support windows resources 👍 See [bazelbuild/rules_cc#131](https://github.com/bazelbuild/rules_cc/pull/131)

All right, how to use it? Let's say I have the simplest `.rc` file defining an icon `my_icon.ico` in the same directory:
```
my_icon ICON "my_icon.ico"
```
How should I compile the `.rc` file with bazel and link the generated object file to `cc_binary`?

Thanks.

**Comment by HelcioM** [comment]
Hello everyone! So there is no direct support for the **msvc rc.exe compiler**? My understanding is that it produce a corresponding .rs binary file and it is then linked to the application.

C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional>where is rc
C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64\rc.exe. T
